### PR TITLE
Feat: added comment support and formatted code

### PR DESCRIPTION
**Pull request by Abb1x**
I added one-line comments support!

### CODE REVIEW

File: src/reader.c [hunk 1]
```diff
@@ -67,58 +67,72 @@ tokenize(const char *content)
 	vec_init(&tokens);
 	vec_init(&buffer);
 
+	bool is_comment = false;
 	for (i = 0; i < strlen(content); i++)
 	{
-		switch (content[i])
+
+		if (content[i] == ';')
```
Comment (Keyboard-Slayer) [inline src/reader.c:74]: You shouldn't delete the comment during the lexing phase, you should rather put it in one block of comment

for example if I follow your example :  
```scheme
(+ 1 2 3) ;  hello comment world!
```

the ast will look like this

`{"+", "1", "2", "3", ";  hello comment world!"};`

File: tests/test_reader.c [hunk 2]
```diff
@@ -29,4 +29,21 @@ read_str_test(void **state)
...
+{
+	vec_str_t tokens;
+	char *t;
+	int i;
+	char *expected[] = { "(","+","1","2","3",")"};
+
+	UNUSED(state);
+	tokens = tokenize("(+ 1 2 3) ; hello comment world!\n");
+
+	vec_foreach(&tokens, t, i) { assert_string_equal(expected[i], t); }
```
Comment (Keyboard-Slayer) [inline tests/test_reader.c:45]: I made a mistake on how I did the first test, I'll at(@) you when I'll fix it. 

I'm looping on the result and not on the template

File: tests/test_reader.c [hunk 3]
```diff
@@ -29,4 +29,21 @@ read_str_test(void **state)
...
+void
+read_str_test_comment(void **state)
+{
+	vec_str_t tokens;
+	char *t;
+	int i;
+	char *expected[] = { "(","+","1","2","3",")"};
+
+	UNUSED(state);
+	tokens = tokenize("(+ 1 2 3) ; hello comment world!\n");
```
Comment (Keyboard-Slayer) [inline tests/test_reader.c:43]: You should write test on other configurations such as:

```scheme 
; test \\n (+ 1 2 3)
```

```scheme 
; test
(+ 1 2 3) 
```

```scheme 
(+ 1 2 3)
; test
```

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: src/reader.c (hunk 1 above)
Comment (Abb1x) [reply src/reader.c:74]: What's the point of not removing them? They're ignored anyway

Comment (Keyboard-Slayer) [reply src/reader.c:74]: It's easier for testing. If we make mistakes we can troubleshot it easily.

**Comment by Abb1x** [comment]
:sweat:

**Comment by Keyboard-Slayer** [comment]
Well, you went trough @Keyboard-Slayer 's coding style *sueur*
And you survived :sweat_smile:

**Pull request merged**